src/utils/componentSwapData.ts

```diff
@@ -39,72 +46,35 @@ export async function getIssuanceComponentSwapData(
-  console.log(issuanceModule)
   const issuance = new Contract(issuanceModule.address, IssuanceAbi, provider)
-  // TODO: check returns
-  const [
-    issuanceComponents, // cDAI, cUSDC and cUSDT, in that order
-    issuanceUnits,
-  ] = await issuance.getRequiredComponentIssuanceUnits(
-    indexToken,
-    indexTokenAmount
-  )
-
-  const indexTokenMix = getIndexTokenMix(indexTokenSymbol)
-  console.log(indexTokenMix)
-  console.log(
-    issuanceComponents,
-    issuanceUnits.map((unit: BigNumber) => unit.toString())
-  )
-
-  // const cDaiExchangeRate = await getExchangeRate(
-  //   '0x5d3a536E4D6DbD6114cc1Ead35777bAB948E3643',
-  //   18,
-  //   provider
-  // )
-  // const cUsdcExchangeRate = await getExchangeRate(
-  //   '0x39AA39c021dfbaE8faC545936693aC917d5E7563',
-  //   6,
-  //   provider
-  // )
-  // const cUsdtExchangeRate = await getExchangeRate(
-  //   '0xf650C3d88D12dB855b8bf7D11Be6C55A4e07dCC9',
-  //   6,
-  //   provider
-  // )
-
-  // const requiredDai = issuanceUnits[0].mul(cDaiExchangeRate)
-  // const requiredUsdc = issuanceUnits[1].mul(cUsdcExchangeRate)
-  // const requiredUsdt = issuanceUnits[2].mul(cUsdtExchangeRate)
-
-  // console.log(requiredDai.toString())
-  // console.log(requiredUsdc.toString())
-  // console.log(requiredUsdt.toString())
-
-  // const buyUnderlyingAmountDai =
-  //   indexTokenMix !== IndexTokenMix.UNWRAPPED_ONLY
-  //     ? requiredDai
-  //     : issuanceUnits[0]
-  // // cUSDC is only used in test case WRAPPED_ONLY
-  // const buyUnderlyingAmountUsdc =
-  //   indexTokenMix === IndexTokenMix.WRAPPED_ONLY
-  //     ? requiredUsdc
-  //     : issuanceUnits[1]
-  // const buyUnderlyingAmountUsdt =
-  //   indexTokenMix !== IndexTokenMix.UNWRAPPED_ONLY
-  //     ? requiredUsdt
-  //     : issuanceUnits[2]
-
-  const swapData = issuanceComponents.map(
-    (component: string, index: number) => {
-      const underlyingERC20 = getUnderlyingErc20(component)
-      return {
-        underlyingERC20,
-        buyUnderlyingAmount: issuanceUnits[index],
-        dexData: getStaticIssuanceSwapData(inputToken, underlyingERC20),
-      }
+  const [issuanceComponents, issuanceUnits] =
+    await issuance.getRequiredComponentIssuanceUnits(
+      indexToken,
+      indexTokenAmount
+    )
+  const underlyingERC20sPromises: Promise<WrappedToken>[] =
+    issuanceComponents.map((component: string) =>
+      getUnderlyingErc20(component, provider)
+    )
+  const wrappedTokens = await Promise.all(underlyingERC20sPromises)
+  const swapData = issuanceComponents.map((_: string, index: number) => {
+    const wrappedToken = wrappedTokens[index]
+    const underlyingERC20 = wrappedToken.underlyingErc20
+    const buyUnderlyingAmount = fromWei(
+      issuanceUnits[index],
+      wrappedToken.decimals,
+      underlyingERC20.decimals
+    )
+      // FIXME: add slippage? Although right it seems to not work with slippage set
+      // in tests (0.5%)
+      .mul(101)
+      .div(100)
```

Comment:
Will fix (by passing slippage into the function) - once it's clear what to do hear or if it's really needed.

Reply:
Note that right now there is no slippage for redemptions as they haven't been tested with the integration tests (need to be able to mint MMI first9